docs/recitations/reci3-deployment.md

```diff
@@ -0,0 +1,180 @@
+
+  
+
+**Continuous Integration (CI)** refers to the practice of automatically building and testing the codebase whenever new changes are introduced. This involves integrating code changes from multiple developers into a shared repository and running automated tests to ensure that the changes do not break the existing functionality.
+
+  
+
+**Continuous Deployment/Delivery (CD)** refers to the practice of automatically deploying code changes to the production environment after they have been tested and approved. Continuous Deployment involves automatically deploying changes to the production environment as soon as they pass automated tests, while Continuous Delivery involves deploying changes to a staging environment for further testing and approval before being deployed to production.
+
+  
+
+Together, CI/CD enables software teams to rapidly and safely develop, test, and deploy changes to the codebase, improving the speed and quality of software delivery while reducing the risk of errors and downtime.
+
+  
+
+There's quite a lot to learn about CI/CD. If you're interested in learning more, [click here!](https://about.gitlab.com/topics/ci-cd/)
+
+  
+
+## Task 0: Setting up your repository
+
+  
+
+For this recitation, we'll be working with a basic web app built on Next.js that responds to HTTP requests. Your job on Tuesday during the _Extreme Startup_ game will be to parse these requests and figure out what the correct response is. Today we'll focus on setting up the app so that everything is in place for the game.
+
+  
+
+The steps to do this are as follows:
+
+!!! Warning
+	Do not use codespaces for this task, and the rest of the recitation. Everything must be done locally.
+
+1. Fork this repo: [https://github.com/CMU-313/basic-web-app](https://github.com/CMU-313/basic-web-app).
+
+2. Follow the instructions in the readme to run and test the development server locally.
+  
+
+## Task 1: Continuous Deployment
+
+To start off, you'll be deploying the app you just forked on two cloud platforms - Render and Google Cloud Platform. The goal here is to make your app accessible over the internet via two platforms. You'll also be setting up continuous deployment such that both of these deployments are updated with the latest version of the app, whenever you push code to the main branch of your repository.
+
+### Task 1A: Deploy on Render
+  
+1. Create an account on [Render](https://render.com)
+2. Create a new Web Service
+3. Choose "Build and deploy from a Git repository"
+4. Click on Configure account under GitHub, to give Render access to your GitHub repositories - this is necessary to setup continuous deployment.
+5. Connect your basic-web-app fork to the Web Service you just created
+6. Give your web service a name
```

Comment:
Make this name unique (e.g., `<andrewid>-313`) since this becomes the deployment domain.